kafka-connect-upload-source/src/main/java/org/radarbase/connect/upload/UploadSourceTask.kt

```diff
@@ -95,17 +95,19 @@ class UploadSourceTask : SourceTask() {
         logger.info("Polling new records...")
         val records: List<RecordDTO> = try {
             uploadClient.pollRecords(PollDTO(1, converters.keys)).records
-        } catch (exe: Exception) {
-            logger.info("Could not successfully poll records. Waiting for next polling...")
+        } catch (exe: Throwable) {
+            logger.error("Could not successfully poll records. Waiting for next polling...")
             nextPoll = failedPollInterval.fromNow()
-            return emptyList()
+            return null
         }
 
         nextPoll = pollInterval.fromNow()
 
         logger.info("Received ${records.size} records at $nextPoll")
 
-        return records.flatMap { record -> processRecord(record) ?: emptyList() }
+        return records
+                .flatMap { record -> processRecord(record) ?: emptyList() }
+                .takeIf { it.isNotEmpty() }
```

Comment:
Do we need this line? why not `records.flatMap { record -> processRecord(record) }`

Reply:
`flatMap` needs a non-null result, even if it is empty. The function should return null instead of empty.